tobago-core/src/main/java/org/apache/myfaces/tobago/application/TobagoFacesMessage.java

```diff
@@ -0,0 +1,18 @@
+package org.apache.myfaces.tobago.application;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class TobagoFacesMessage extends LabelValueExpressionFacesMessage {
+
+    private final Map<Object, Object> dataAttributes = new HashMap<>();
+
+    public Map<Object, Object> getDataAttributes() {
```

Comment:
DataAttributes or just Attributes in FacesMessage?

Reply:
I would stick with DataAttributes. After all, only data-xxx attributes are rendered. Otherwise you should also render other attributes